msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -3381,6 +3405,72 @@ else if(getUserId()==null&&this.forumManager.getAnonRole()==true){
     return decoTopic;
   }
 
+	public void setAnonymousManager(AnonymousManager anonymousManager)
+	{
+		this.anonymousManager = anonymousManager;
+	}
+
+	public boolean isAnonymousEnabled()
+	{
+		return anonymousManager.isAnonymousEnabled();
+	}
+
+	public boolean isSiteHasAnonymousTopics()
+	{
+		return forumManager.isSiteHasAnonymousTopics(getSiteId());
+	}
+
+	/**
+	 * Determines whether the current user should see anonymous IDs in the context of the specified topic.
+	 * @param topic
+	 * @param true if the topic is anonymous and revealIDsToRoles is disabled or
+	 * revealIDsToRoles is enabled, but the user doesn't have the permission to identify users in this topic
+	 */
+	private boolean isUseAnonymousId(Topic topic)
+	{
+		if (topic == null)
+		{
+			throw new IllegalArgumentException("isUseAnonymousId invoked with null topic");
+		}
+
+		if (!isAnonymousEnabled())
+		{
+			return false;
+		}
+
+		// if topic.postAnonymous
```

Comment:
Ah, sorry didn't see the end of it. I'm all in favour of more readable code and agree it's hard to read. I'd write it out more fully in Java so that it's more self documenting.

    boolean revealAnonIds = topic.getRevealIDsToRoles() && uiPermissionsManager.isIdentifyAnonAuthors(topic);
    return topic.getPostAnonymous() && ! revealAnonIds;

That way it's easier if anyone ever has to debug it and doesn't have to trust that the comments and code are in sync. Although having written that I'm not sure my suggestion is much clearer.